src/libos/src/net/socket/unix/stream/address_space.rs

```diff
@@ -88,17 +107,39 @@ impl AddressSpace {
         space.remove(&key);
     }
 
-    fn get_space(&self, addr: &Addr) -> SgxMutexGuard<'_, BTreeMap<String, Option<Arc<Listener>>>> {
+    fn get_space(
+        &self,
+        addr: &Addr,
+    ) -> SgxMutexGuard<'_, BTreeMap<AddressSpaceKey, Option<Arc<Listener>>>> {
         match addr {
-            Addr::File(unix_path) => self.file.lock().unwrap(),
+            Addr::File(unix_path, inode) => self.file.lock().unwrap(),
             Addr::Abstract(path) => self.abstr.lock().unwrap(),
         }
     }
 
-    fn get_key(addr: &Addr) -> String {
+    fn get_key(addr: &Addr) -> AddressSpaceKey {
```

Comment:
Rewrite this function

```
fn get_key(addr: &Addr) -> Option<AddressSpaceKey> {
    trace!("addr = {:?}", addr);
    match addr {
        Addr::File(unix_path, inode_num) if inode_num.is_some() => {
            Some(AddressSpaceKey::from_inode(inode_num.unwrap()))
        }
        Addr::File(unix_path, _) => {
            let inode = {
                let file_path = unix_path.absolute();
                let current = current!();
                let fs = current.fs().read().unwrap();
                fs.lookup_inode(&file_path)
            };
            if let Ok(inode) = inode {
                Some(AddressSpaceKey::from_inode(inode.metadata().unwrap().inode));
            } else {
                None
            }
        }
        Addr::Abstract(path) => Some(AddressSpaceKey::from_path(addr.path_str().to_string())),
    }
}
```

Reply:
Thank you for the review. I think this is more concise and clear. Done.